provider-blackduck/src/main/java/com/synopsys/integration/alert/provider/blackduck/processor/message/service/BlackDuckMessageBomComponentDetailsCreatorFactory.java

```diff
@@ -27,8 +29,10 @@ public BlackDuckMessageBomComponentDetailsCreatorFactory(
         this.policyDetailsCreator = policyDetailsCreator;
     }
 
-    public BlackDuckMessageBomComponentDetailsCreator createBomComponentDetailsCreator(BlackDuckApiClient blackDuckApiClient) {
-        return new BlackDuckMessageBomComponentDetailsCreator(blackDuckApiClient, vulnerabilityDetailsCreator, policyDetailsCreator);
+    public BlackDuckMessageBomComponentDetailsCreator createBomComponentDetailsCreator(BlackDuckServicesFactory blackDuckServicesFactory) {
+        BlackDuckApiClient blackDuckApiClient = blackDuckServicesFactory.getBlackDuckApiClient();
+        PolicyRuleService policyRuleService = blackDuckServicesFactory.createPolicyRuleService();
```

Comment:
Instead of passing in BlackDuckServicesFactory we generally want to pass in BlackDuckApiClient and PolicyRuleService. Since we don't use the object to perform any function or modify it in anyway we don't really need the factory. It should also make testing simpler.

Reply:
The reasoning for this was that the extractors would then need to also call the BlackDuckServicesFactory.createPolicyRuleService() in each of the extractor classes such as VulnerabiliyNotificationMessageExtractor. Rather than call that 4 times across those classes we only call it once here.